File: QBRTCDemo/Models/QBUUser+SVUser.m
//
//  QBUUser+SVUser.m
//  QBRTCDemo
//
//  Created by Anton Sokolchenko on 11/17/15.
//  Copyright © 2015 anton. All rights reserved.
//

#import "QBUUser+SVUser.h"
#import "SVUser.h"

@implementation QBUUser (SVUser)

+ (instancetype)userWithSVUser:(SVUser *)svuser {
	QBUUser *qbuser = [QBUUser user];
	qbuser.ID = svuser.ID.unsignedIntegerValue;
	qbuser.login = svuser.login;
	qbuser.password = svuser.password;
	return qbuser;
}

@end

File: QBRTCDemo/Modules/Models/QBUser+SVUser/QBUUser+SVUser.swift
//
//  QBUUser+SVUser.swift
//  QBRTCDemo
//
//  Created by Anton Sokolchenko on 3/13/19.
//  Copyright © 2019 Anton Sokolchenko. All rights reserved.
//

import Foundation
import Quickblox

extension QBUUser {
	class func fromSVUser(svuser: SVUser) -> QBUUser {
		let qbuser = QBUUser()
		if let svuserID = svuser.ID {
			qbuser.id = UInt(svuserID)
		}
		qbuser.login = svuser.login
		qbuser.password = svuser.password
		qbuser.tags = NSMutableArray(array: svuser.tags ?? [])
		qbuser.fullName = svuser.fullName
		return qbuser
	}
}
